distributed/tests/test_client.py

```diff
@@ -6600,6 +6600,33 @@ def log_scheduler(dask_scheduler):
     assert events[1][1] == ("alice", "bob")
 
 
+@gen_cluster(client=True, nthreads=[])
+async def test_log_event_multiple_clients(c, s):
+    c2 = await Client(s.address, set_as_default=False, asynchronous=True)
+
+    received_events = []
+
+    def get_event_handler(handler_id):
+        def handler(event):
+            received_events.append((handler_id, event))
+
+        return handler
+
+    c.subscribe_topic("test-topic", get_event_handler(1))
+    c2.subscribe_topic("test-topic", get_event_handler(2))
+
+    while len(s.event_subscriber["test-topic"]) != 2:
+        await asyncio.sleep(0.01)
+
+    await c.log_event("test-topic", {})
+
+    while len(received_events) < 2:
+        await asyncio.sleep(0.01)
+
+    assert len(received_events) == 2
+    assert {ev[0] for ev in received_events} == {1, 2}
```

Comment:
From what I understand from #7013, having a non-subscribed client around caused a `ValueError before. Let's add it to the test case then:

```suggestion
        async with Client(s.address, asynchronous=True) as c2, Client(
            s.address, asynchronous=True
        ) as c3:
            received_events = []

            def get_event_handler(handler_id):
                def handler(event):
                    received_events.append((handler_id, event))

                return handler

            c.subscribe_topic("test-topic", get_event_handler(1))
            c2.subscribe_topic("test-topic", get_event_handler(2))

            while len(s.event_subscriber["test-topic"]) != 2:
                await asyncio.sleep(0.01)

            await c.log_event("test-topic", {})

            import time

            time.sleep(2)
            while len(received_events) < 2:
                await asyncio.sleep(0.01)

            assert len(received_events) == 2
            assert {handler_id for handler_id, _ in received_events} == {1, 2}
            assert "ValueError" not in logger.getvalue()
```
In this suggestion, I have made the following adjustments:
1. Add the non-subscribed client
2. Use an async context manager to ensure teardown of the clients
3. Capture the client logs to assert that the `ValueError` from #7013 is not logged
4. Stylistic nit to the set comparison

Reply:
Thanks - Looks good, but do we need this part? 

```
import time

time.sleep(2)
```

Reply:
Whoops, good catch, that snuck in during copy+paste. I've adjusted the suggestion.